### PR TITLE
thunderbird-bin: adopt common update script

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

firefox-bin now uses the common update script (#20844). This patch brings it to thunderbird-bin.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc: @garbas

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix [hunk 1]
```diff
@@ -141,6 +146,72 @@ stdenv.mkDerivation {
...
+            }
+        EOF
+          done
+        done
+        cat >> $tmpfile <<EOF
+          ];
+        }
+        EOF
+
+        cat $tmpfile > sources.nix
```
Comment (Mic92) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: cp?

Comment (taku0) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: Changed to `mv`.

**Comment by garbas** [comment]
@taku0 it looks like we could use almost the same update script for both firefox-bin and thunderbird-bin.

**Comment by garbas** [comment]
I've used the same update script for both firefox-bin and thunderbird-bin in #21208.

**Pull request closed without merging**